### PR TITLE
Aave v3 import fix

### PR DESCRIPTION
**Pull request by iri031**
Aave V3 import with Collateral. 
- Function `importAaveWithCollateral` takes array of bools `enableCollateral` as param.
- Will enable only selected tokens as collateral while importing the position.
- [x] Implementation
- [x] Deployment
- [x] Add to SDK https://github.com/Instadapp/dsa-connect/pull/168
- [x] Enable
- [ ] Update docs

### CODE REVIEW

File: contracts/polygon/connectors/aave/v3-import/helpers.sol [hunk 1]
```diff
@@ -250,6 +250,34 @@ contract AaveHelpers is Helper {
...
+						address(this),
+						_amt
+					),
+					"allowance?"
+				);
+
+				if (!getIsColl(tokens[i], address(this))) {
+					aave.setUserUseReserveAsCollateral(tokens[i], true);
+				}
+			}
```
Comment (thrilok209) [inline contracts/polygon/connectors/aave/v3-import/helpers.sol]:
```suggestion
			if (amts[i] > 0) {
				uint256 _amt = amts[i];
				require(
					atokenContracts[i].transferFrom(
						userAccount,
						address(this),
						_amt
					),
					"allowance?"
				);

				if (!getIsColl(tokens[i], address(this))) {
					aave.setUserUseReserveAsCollateral(tokens[i], colEnable[i]);
				}
			}
```

@Richa-iitr This is the main change we needed. We will be taking atoken, but to enable atoken as collateral or not will be based on `colEnable`

File: contracts/polygon/connectors/aave/v3-import/main.sol [hunk 2]
```diff
@@ -92,6 +92,88 @@ contract AaveV3ImportResolver is AaveHelpers {
...
+	function _importAaveWithCollateral(address userAccount, ImportInputData memory inputData, bool[] memory enableCollateral)
+		internal
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		require(
+			AccountInterface(address(this)).isAuth(userAccount),
+			"user-account-not-auth"
+		);
+
+		require(inputData.supplyTokens.length > 0, "0-length-not-allowed");
```
Comment (thrilok209) [inline contracts/polygon/connectors/aave/v3-import/main.sol:104]: @Richa-iitr Can you check if `enableCollateral` length eq to `supplyTokens` length in another require statement?

File: contracts/polygon/connectors/aave/v3-import/helpers.sol (hunk 1 above)
Comment (iri031) [reply contracts/polygon/connectors/aave/v3-import/helpers.sol]: got it.

File: contracts/polygon/connectors/aave/v3-import/main.sol (hunk 2 above)
Comment (iri031) [reply contracts/polygon/connectors/aave/v3-import/main.sol:104]: okay

File: contracts/polygon/connectors/aave/v3-import/main.sol [hunk 3]
```diff
@@ -105,6 +187,21 @@ contract AaveV3ImportResolver is AaveHelpers {
...
+	function importAaveWithCollateral(address userAccount, ImportInputData memory inputData, bool[] memory enableCollateral)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		(_eventName, _eventParam) = _importAaveWithCollateral(userAccount, inputData, enableCollateral);
+	}
 }
 
 contract ConnectV2AaveV3ImportPolygon is AaveV3ImportResolver {
```
Comment (thrilok209) [inline contracts/polygon/connectors/aave/v3-import/main.sol:208]: Update version on name to `Aave-v3-import-v1.1`

File: contracts/polygon/connectors/aave/v3-import/main.sol (hunk 2 above)
Comment (iri031) [reply contracts/polygon/connectors/aave/v3-import/main.sol:104]: done

File: contracts/polygon/connectors/aave/v3-import/main.sol (hunk 3 above)
Comment (iri031) [reply contracts/polygon/connectors/aave/v3-import/main.sol:208]: done

**Comment by iri031** [comment]
## Deployed connector for Polygon 

**Name** `AAVE-V3-IMPORT-A`
**Addr** [0x34c1dE737E201F9581f37564B492B492288eb7aF](https://polygonscan.com/address/0x34c1dE737E201F9581f37564B492B492288eb7aF#code)
**tx** [here](https://polygonscan.com/tx/0x7ed04cfaa42a6254150cc189c91af5aa0292680cc6d79e317c4c5ab6199cb71e)

## Deployed connector for Avalanche
**Name** `AAVE-V3-IMPORT-A`
**Addr** [0x5c31E2d1c611d062beFdf3883AdE05a32c52b9Cb](https://snowtrace.io/address/0x5c31E2d1c611d062beFdf3883AdE05a32c52b9Cb#code)
**tx** [here](https://snowtrace.io/tx/0xa3a989886d2b175512459c219c026e8a63e6c43914a94ea126490a21a63b8a1d)

## Deployed connector for Arbitrum
**Name** `AAVE-V3-IMPORT-A`
**Addr** [0xDFF1091149bE811aDB2e5FE0fB48DBa8DcE5a755](https://arbiscan.io/address/0xDFF1091149bE811aDB2e5FE0fB48DBa8DcE5a755#code)
**tx**  [here](https://arbiscan.io/tx/0xf1ef52ca74579a6384c4a6723df282e339780bf15c52ec65393cf7dcd37469b9)

## Deployed connector for Fantom 
**Name** `AAVE-V3-IMPORT-A`
**Addr** [0x7cdBD859f2EDA545289378112FD991571d6eb73e](https://ftmscan.com/address/0x7cdBD859f2EDA545289378112FD991571d6eb73e#code)
**tx**  [here](https://ftmscan.com/tx/0x91be3f195bf9b982c26141fd6e9536e736ec50f1538841de636cfa1313aa781d#eventlog)

## Deployed connector for Optimism
**Name** `AAVE-V3-IMPORT-A`
**Addr** [0xDFF1091149bE811aDB2e5FE0fB48DBa8DcE5a755](https://optimistic.etherscan.io/address/0xDFF1091149bE811aDB2e5FE0fB48DBa8DcE5a755#code)
**tx** [here](https://optimistic.etherscan.io/tx/0x7b06d64d3fce8b9afc8685cb7eac38b18d5f5fc2a4abf4412f4f0269836544e8)

**Review by pradyuman-verma — APPROVED** [review]

**Pull request merged**